src/utils/number.spec.ts

```diff
@@ -97,3 +98,32 @@ describe("sanitizeNumberString", () => {
     expect(sanitizeNumberString(crazyExponentialString)).toBe("-2.1e53109");
   });
 });
+
+describe("BigDecimal", () => {
+  it("handles precise/large numbers and arbitrary-precision arithmetic", () => {
+    const subtract = new BigDecimal("0.3").subtract("0.1").toString();
+    expect(subtract).toBe("0.2");
+
+    const add = new BigDecimal(Number.MAX_SAFE_INTEGER.toString()).add("1").toString();
+    expect(add).not.toBe(Number.MAX_SAFE_INTEGER.toString());
```

Comment:
Could you add an additional assertion that the resulting sum is 1 higher than the original value too?